src/graph/planner/match/MatchPlanner.cpp

```diff
@@ -83,11 +83,19 @@ Status MatchPlanner::connectMatchPlan(SubPlan& queryPlan, MatchClauseContext* ma
   }
   if (!intersectedAliases.empty()) {
     if (matchCtx->isOptional) {
+      // connect LeftJoin match filter
+      if (matchCtx->where != nullptr) {
```

Comment:
What will happen when filter reference variable outside current match cluase if you don't delay the filter

Reply:
```
MATCH (a)--(b)--(c) 
OPTIONAL MATCH (a)--(b)--(d) WHERE d.prop>c.prop
RETURN count(*) AS count
```
This statement is currently not fixed, `count` will return 0.

Reply:
If this hotfix introduces a new bug , it's not necessary to merge. WDYT? @CPWstatic

Reply:
Do you mean the previous hotfix or this one?